solr/core/src/java/org/apache/solr/handler/component/DebugComponent.java

```diff
@@ -115,6 +116,12 @@ public void process(ResponseBuilder rb) throws IOException {
         info.add("facet-debug", fdebug.getFacetDebugInfo());
       }
 
+      ReverseSearchDebugInfo rsdebug =
+          (ReverseSearchDebugInfo) (rb.req.getContext().get(ReverseSearchDebugInfo.KEY));
+      if (rsdebug != null) {
+        info.add("reverse-search-debug", rsdebug.getReverseSearchDebugInfo());
+      }
+
```

Comment:
If `DebugComponent.java` had a method something like this

```
protected void addCustomInfo(ResponseBuilder rb,  NamedList<Object> info) {
  FacetDebugInfo fdebug = (FacetDebugInfo) (rb.req.getContext().get("FacetDebugInfo"));
  if (fdebug != null) {
    info.add("facet-trace", fdebug.getFacetDebugInfo());
  }
  fdebug = (FacetDebugInfo) (rb.req.getContext().get("FacetDebugInfo-nonJson"));
  if (fdebug != null) {
    info.add("facet-debug", fdebug.getFacetDebugInfo());
  }
}
```

then a `solr/modules/monitor` (or indeed any other custom plugin) could extend the class and override the method.

Reply:
Added https://github.com/apache/solr/pull/2382/commits/44eea79ba27041555ee5070ad4b6c3e0f6c3bb2a commit to explore that. WDYT?

Reply:
I like it. It would be nice it you didn't have to use inheritance because it makes "layering" debug components challenging (i.e. if the debug component could iterate over other customInfoProviders to collect custom debug metrics..) . But that's a hypothetical problem since nothing that requires this exists yet and the DI mechanism in solr is a bit painful right now so doing composition is probably messier.

One question I have is why not keep the `ReverseSearchDebugComponent` in the default component list for the reverseSearchHandler? The searchHandler includes `DebugComponent` in its default components so I would think we'd want to keep that consistent.

Reply:
I suppose an alternative way would be to introduce some normalized List of `CustomInfoProviders` in the request context .. the inteface could like like:

```
interface CustomDebugInfoProvider {
    SimpleOrderedMap<Object> asMap()
    String name();

}
```

and you could fetch it via:

```
List<CustomDebugInfoProvider > customInfos = (List<CustomDebugInfoProvider >) request.getContext().get("customDebugInfoProvider")
for (var customInfo: customInfos) {
    info.add(customInfo.name(), customInfo.asMap());
}
```

LEt me know what you think .. I can also make a commit to clarify..

Reply:
> ... an alternative way ...

Yeah, I had no particular way or implementation in mind as such, only that as a concept `solr/modules/foobar` components leaving 'footprint' in `solr/core DebugComponent.java` is perhaps undesirable and that 'solving' it would also benefit custom components in general, something along those lines.

Reply:
> One question I have is why not keep the `ReverseSearchDebugComponent` in the default component list for the reverseSearchHandler? The searchHandler includes `DebugComponent` in its default components so I would think we'd want to keep that consistent.

The `QueryComponent` and `DebugComponent` are implicit components, so that makes it easier to include them in the defaults I guess. No strong views on including or not including `ReverseSearchDebugComponent` in the default components but I wanted to also explore/demonstrate that the code works even if the debug component is _not_ configured.

https://github.com/apache/solr/pull/2382/commits/3e7a53c8095f4a33c5ce194abbfb27d154a70571 (somewhat subjectively of course) removes the sink factory since there's only one type of sink now, this (perhaps) makes the code easier to understand and it also allowed `rb.isDebug()` qualification for the collection of the debug info.